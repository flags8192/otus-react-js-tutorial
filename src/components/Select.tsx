import React from 'react';
import styled from '@emotion/styled';
import { FC } from 'react';

const Root = styled.select`
  height: 30px;
  margin: auto 20px;
`;

export interface ISelect {
  onSelect?: (level: React.ChangeEvent<HTMLSelectElement>) => void;
}
const Select: FC<ISelect> = ({ onSelect }) => {
  return (
    <Root role="select" name="select" onChange={onSelect}>
      <option role="option" value="1" defaultValue="1">
        Падаван
      </option>
      <option role="option" value="2">
        Джедай
      </option>
      <option role="option" value="3">
        Мастер-джедай
      </option>
      <option role="option" value="4">
        Член Совета Силы
      </option>
    </Root>
  );
};

export default Select;
