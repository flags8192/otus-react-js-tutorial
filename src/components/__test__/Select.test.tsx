import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Select from '@src/components/Select';

describe('Select', () => {
  it('render <Select>', () => {
    const handleSelect = jest.fn();
    const { asFragment } = render(<Select onSelect={handleSelect} />);
    expect(asFragment()).toMatchSnapshot();
    let select = screen.getByRole(/select/gi);
    expect(select).toBeInTheDocument();
  });

  it('select on <Select>', () => {
    const handleSelect = jest.fn();
    render(<Select onSelect={handleSelect} />);
    const select = screen.getByRole(/select/gi);
    userEvent.selectOptions(select, '1');
    const option1 = screen.getByRole('option', { name: 'Падаван' }) as HTMLOptionElement;
    expect(option1.selected).toBe(true);
    const option2 = screen.getByRole('option', { name: 'Джедай' }) as HTMLOptionElement;
    expect(option2.selected).toBe(false);
    const option3 = screen.getByRole('option', { name: 'Мастер-джедай' }) as HTMLOptionElement;
    expect(option3.selected).toBe(false);
    const option4 = screen.getByRole('option', { name: 'Член Совета Силы' }) as HTMLOptionElement;
    expect(option4.selected).toBe(false);
  });
});
