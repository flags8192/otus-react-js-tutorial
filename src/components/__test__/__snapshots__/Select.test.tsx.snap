// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select render <Select> 1`] = `
<DocumentFragment>
  <select
    class="css-tv5zn1"
    name="select"
    role="select"
  >
    <option
      role="option"
      value="1"
    >
      Падаван
    </option>
    <option
      role="option"
      value="2"
    >
      Джедай
    </option>
    <option
      role="option"
      value="3"
    >
      Мастер-джедай
    </option>
    <option
      role="option"
      value="4"
    >
      Член Совета Силы
    </option>
  </select>
</DocumentFragment>
`;
